### PR TITLE
[fix] Disabled, unimplemented, and unknown opcodes should throw an error

### PR DESCRIPTION
**Pull request by adrianvrj**
- [x] issue #109 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Added the necessary logic to return errors without executing opcodes inside a if statement when is falase

### CODE REVIEW

File: src/opcodes/opcodes.cairo [hunk 1]
```diff
@@ -264,4 +265,176 @@ pub mod Opcode {
             _ => utils::not_implemented(ref engine)
         }
     }
+
+    pub fn check_opcode(opcode: u8, ref engine: Engine) -> Result<(), felt252> {
```
Comment (b-j-roberts) [inline src/opcodes/opcodes.cairo:269]: I think this implementation of this function will be a bit slow, and is hard to read and maintain.

I'd recommend using an if statement with `||` expressions instead for all the possible disabled opcodes. Also, calling the function something like `is_opcode_disabled` would be more accurate.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: src/opcodes/opcodes.cairo (hunk 1 above)
Comment (adrianvrj) [reply src/opcodes/opcodes.cairo:269]: Working on it!

File: src/opcodes/opcodes.cairo [hunk 2]
```diff
@@ -364,4 +365,26 @@ pub mod Opcode {
             _ => utils::not_implemented(ref engine)
         }
     }
+
+    pub fn is_opcode_disabled(opcode: u8, ref engine: Engine) -> Result<(), felt252> {
+        if opcode == 126
```
Comment (b-j-roberts) [inline src/opcodes/opcodes.cairo:370]: Could you use the opcode names here now to make them easier to read? For example, this line would be something like `opcode == OP_CAT`.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: src/opcodes/opcodes.cairo (hunk 2 above)
Comment (adrianvrj) [reply src/opcodes/opcodes.cairo:370]: Sure!

**Review by b-j-roberts — APPROVED** [review]
Nice work! Thanks for the quick changes.

**Comment by adrianvrj** [comment]
> Nice work! Thanks for the quick changes.

🫡🫡🫡

**Pull request merged**